web-ui/src/analytics/TableSqlPrograms.tsx

```diff
@@ -0,0 +1,182 @@
+// Display a table with all SQL programs.
+//
+// Can edit name and description directly in the table.
+// Also display status of the program.
+
+import { useRouter } from 'next/router'
+import { useState, useCallback } from 'react'
+import Box from '@mui/material/Box'
+import Card from '@mui/material/Card'
+import Typography from '@mui/material/Typography'
+import { GridColumns, GridRenderCellParams, useGridApiRef } from '@mui/x-data-grid-pro'
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query'
+import { match, P } from 'ts-pattern'
+
+import CustomChip from 'src/@core/components/mui/chip'
+import { ProjectService } from 'src/types/manager/services/ProjectService'
+import { ProjectDescr } from 'src/types/manager/models/ProjectDescr'
+import { ProjectStatus } from 'src/types/manager/models/ProjectStatus'
+import { CancelError, UpdateProjectRequest, UpdateProjectResponse } from 'src/types/manager'
+import EntityTable from 'src/components/table/EntityTable'
+import useStatusNotification from 'src/components/errors/useStatusNotification'
+
+const getStatusObj = (status: ProjectStatus) =>
+  match(status)
+    .with({ SqlError: P._ }, () => {
+      return { title: 'SQL Error', color: 'error' as const }
```

Comment:
as const is necessary for type safety

Reply:
Not having a return also makes the type-checker complain. I remember I fought with this before but couldn't get it to accept it without return :/

Reply:
I don't understand why const is required. Can you try without it?

Reply:
To remove return you heed to remove the enclosing block

Reply:
I did remove enclosing block but that didnt work... The as const is needed because the type for color is ThemeColor which is something like 'abc' | 'def'

I had this type error for a while until I found a comment in an issue in the typescript repo that said you need to add 'as const'. I tried to find the issue again yesterday but couldnt find it anymore.